### PR TITLE
Migrate project to ViewBinding

### LINKED ISSUE #16: Migrate to ViewBinding in the project
WhatTheStack started as a small project to only display very basic information on a crash. `findViewById` sufficed for this purpose.

However, the complexity of the error screen has increased since then, and is going to continue in that direction. Therefore it makes sense to now refactor the project to migrate to [ViewBinding](https://developer.android.com/topic/libraries/view-binding) instead.

Tasks:
- Enable ViewBinding in the project's build configuration
- Migrate to ViewBinding in the sample app
- Migrate to ViewBinding in the library

### PR DESCRIPTION
**Pull request by shivam21**
* Enable ViewBinding in the project's build configuration
* Migrate to ViewBinding in the sample app
* Migrate to ViewBinding in the library
* Fixes #16

### CODE REVIEW

File: app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt [hunk 1]
```diff
@@ -2,18 +2,20 @@ package com.haroldadmin.crashyapp
...
 import androidx.appcompat.app.AppCompatActivity
-import kotlinx.android.synthetic.main.activity_main.*
+import com.haroldadmin.crashyapp.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        val activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
```
Comment (haroldadmin) [inline app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt]: A good convention to follow for binding variables is to just call them `binding`. Their context is evident from their parent Activity/Fragment class, so there's no need to include their name in the variable.

File: app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt [hunk 2]
```diff
@@ -2,18 +2,20 @@ package com.haroldadmin.crashyapp
...
-        crashButton.setOnClickListener {
+        activityMainBinding.crashButton.setOnClickListener {
             throw BecauseICanException()
         }
     }
 }
 
-private class BecauseICanException : Exception("This exception is thrown purely because it can be thrown")
+private class BecauseICanException :
+    Exception("This exception is thrown purely because it can be thrown")
```
Comment (haroldadmin) [inline app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt]: Please refrain from unnecessary formatting changes.

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt [hunk 3]
```diff
@@ -7,11 +7,9 @@ import android.content.Intent
...
 import android.view.View
-import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
-import androidx.appcompat.widget.AppCompatTextView
-import com.google.android.material.button.MaterialButton
 import com.google.android.material.snackbar.Snackbar
+import com.haroldadmin.whatthestack.databinding.ActivityWhatTheStackBinding
 import dev.chrisbanes.insetter.Insetter
 import dev.chrisbanes.insetter.Side
 import kotlinx.android.synthetic.main.activity_what_the_stack.*
```
Comment (haroldadmin) [inline what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt]: Remove the synthetics import, since its not required.

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt [hunk 4]
```diff
@@ -28,7 +26,8 @@ class WhatTheStackActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_what_the_stack)
+        val activityWhatTheStackBinding = ActivityWhatTheStackBinding.inflate(layoutInflater)
```
Comment (haroldadmin) [inline what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt]: Please follow the naming convention for binding variables here too.

**Review by haroldadmin — CHANGES_REQUESTED** [review]
Good work! Please address a few small pieces of feedback I have left.

**Comment by haroldadmin** [comment]
For future PRs, please make sure to include a line like `Fixes #<issue-number>` in the description. This links the PR to the issue, and ensures that when the PR is merged the issue will be automatically closed.

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt (hunk 4 above)
Comment (shivam21) [reply what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt]: done

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt (hunk 3 above)
Comment (shivam21) [reply what-the-stack/src/main/java/com/haroldadmin/whatthestack/WhatTheStackActivity.kt]: done

File: app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt (hunk 2 above)
Comment (shivam21) [reply app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt]: done

File: app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt (hunk 1 above)
Comment (shivam21) [reply app/src/main/java/com/haroldadmin/crashyapp/MainActivity.kt]: done

**Review by haroldadmin — APPROVED** [review]
Looks good! Thank you for another contribution 🎉
